test/test_initial_design/test_factorical_design.py

```diff
@@ -0,0 +1,52 @@
+import unittest
+import unittest.mock
+
+import numpy as np
+from ConfigSpace import ConfigurationSpace, UniformFloatHyperparameter,\
+    Constant, CategoricalHyperparameter, OrdinalHyperparameter
+
+from smac.initial_design.factorial_design import FactorialInitialDesign
+
+
+class TestFactorial(unittest.TestCase):
+    def test_factorial(self):
+        def get_uniform_param(name: str):
+            return UniformFloatHyperparameter(name, 0, 1)
+
+        def get_constant_param(name: str):
+            return Constant(name, 0.)
+
+        def get_categorical_param(name: str):
+            return CategoricalHyperparameter(name, choices=["a", "b", "c"])
+
+        def get_ordinal_param(name: str):
+            return OrdinalHyperparameter(name, [8, 6, 4, 2])
+
+        get_params = [
+            get_uniform_param,
```

Comment:
why don't you write 
`UniformFloatHyperparameter(name, 0, 1)`
directly here?

Reply:
Because I can iterate over the param types more easily, does not really matter